src/components/sets/CellSetExpressionPlot.js

```diff
@@ -0,0 +1,253 @@
+import { VegaPlot, VEGA_THEMES, DATASET_NAME } from '../vega';
+import { colorArrayToString } from './utils';
+
+/**
+ * Gene expression histogram displayed as a bar chart,
+ * implemented with the VegaPlot component.
+ * @param {object} props
+ * @param {object[]} props.data The set size data, an array
+ * of objects with properties `value` and `gene`.
+ * @param {string} props.theme The name of the current Vitessce theme.
+ * @param {number} props.width The container width.
+ * @param {number} props.height The container height.
+ * @param {number} props.marginRight The size of the margin
+ * on the right side of the plot, to account for the vega menu button.
+ * By default, 90.
+ * @param {number} props.marginBottom The size of the margin
+ * on the bottom of the plot, to account for long x-axis labels.
+ * By default, 50.
+ */
+export default function CellSetExpressionPlot(props) {
+  const {
+    colors,
+    data,
+    theme,
+    width,
+    height,
+    marginRight = 90,
+    marginBottom = 120,
+  } = props;
+
+  // Manually set the color scale so that Vega-Lite does
+  // not choose the colors automatically.
+  const colorScale = {
+    domain: colors.map(d => d.name),
+    range: colors.map(d => colorArrayToString(d.color)),
+  };
+
+  const plotWidth = clamp(width - marginRight, 10, Infinity);
+  const plotHeight = clamp(height - marginBottom, 10, Infinity);
+
+  const numBands = colors.length;
+  const bandWidth = plotWidth / numBands;
+
+  const spec = {
+    $schema: 'https://vega.github.io/schema/vega/v5.json',
+    description: 'A violin plot showing distributions of expression levels for selected cell sets.',
+    width: plotWidth,
+    height: plotHeight,
+    config: {
+      ...VEGA_THEMES[theme],
+      axisBand: {
+        bandPosition: 1,
+        tickExtra: true,
+        tickOffset: 0,
+      },
+    },
+
+    signals: [
+      { name: 'bandWidth', value: bandWidth },
+      { name: 'width', value: plotWidth },
+      { name: 'height', value: plotHeight },
+      { name: 'trim', value: true },
+    ],
+
+    data: [
+      {
+        name: 'density',
+        source: DATASET_NAME,
+        transform: [
+          {
+            type: 'kde',
+            field: 'value',
+            groupby: ['set'],
+            bandwidth: 0,
+            extent: [0, 100],
+          },
+        ],
+      },
+      {
+        name: 'stats',
+        source: DATASET_NAME,
+        transform: [
+          {
+            type: 'aggregate',
+            groupby: ['set'],
+            fields: ['value', 'value', 'value'],
+            ops: ['q1', 'median', 'q3'],
+            as: ['q1', 'median', 'q3'],
+          },
+        ],
+      },
+    ],
+
+    scales: [
+      {
+        name: 'layout',
+        type: 'band',
+        range: 'width',
+        domain: { data: DATASET_NAME, field: 'set' },
+      },
+      {
+        name: 'yscale',
+        type: 'linear',
+        range: 'height',
+        domain: [0, 100],
```

Comment:
<img width="433" alt="Screen Shot 2021-01-19 at 11 01 23 AM" src="https://user-images.githubusercontent.com/43999641/105059852-b8868900-5a45-11eb-9a96-92b328a66e96.png">

Reply:
I have updated this so that the domain's max value is computed from the data. I think in the long term we also need to explore different gene expression normalization approaches, so I added this issue https://github.com/vitessce/vitessce/issues/844